src/stores/media.ts

```diff
@@ -0,0 +1,431 @@
+         * fetch. Normally, if `shouldPersistMedia` is true, `page` should have been set to 1 by the
+         * previous fetch. But if it wasn't and is still undefined, we set it to 0, and increment it.
+         */
+        page = (state.results[mediaType].page ?? 0) + 1
+        queryParams.page = `${page}`
+      }
+      fetchStartMedia(mediaType)
+      try {
+        const data = await services[mediaType].search(queryParams)
+
+        fetchEndMedia(mediaType)
+        const mediaCount = data.result_count
+        if (!mediaCount) {
+          fetchMediaError({
+            mediaType,
+            errorMessage: `No ${mediaType} found for this query`,
+          })
+          page = undefined
+        }
+        setMedia({
+          mediaType,
+          media: data.results,
+          mediaCount,
+          pageCount: data.page_count,
+          shouldPersistMedia,
+          page,
+        })
+      } catch (error) {
+        await handleMediaError({ mediaType, error })
+      }
+    }
+    /**
+     *
+     */
+    const fetchMediaItem = async (params: {
+      mediaType: SupportedMediaType
+      id: string
+    }) => {
+      const { mediaType } = params
+      try {
+        // TODO: Fix this!
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        state[mediaType] = await services[mediaType].getMediaDetail(params.id)
+      } catch (error: unknown) {
+        state[mediaType] = null
+        if (axios.isAxiosError(error) && error.response?.status === 404) {
+          mediaNotFound(mediaType)
+        } else {
+          await handleMediaError({ mediaType, error })
+        }
+      }
+    }
+    /**
+     *
+     */
+    const handleMediaError = async ({
+      mediaType,
+      error,
+    }: {
+      mediaType: SupportedMediaType
+      error: unknown
+    }) => {
+      let errorMessage
+      if (axios.isAxiosError(error)) {
+        errorMessage =
+          error.response?.status === 500
+            ? 'There was a problem with our servers'
+            : `Request failed with status ${
+                error.response?.status ?? 'unknown'
+              }`
+      } else {
+        errorMessage =
+          error instanceof Error ? error.message : 'Oops! Something went wrong'
+      }
+      fetchMediaError({ mediaType, errorMessage: errorMessage })
+      if (!axios.isAxiosError(error)) {
+        throw new Error(errorMessage)
+      }
+    }
+
+    return {
+      state,
+
+      getItemById,
+      resultItems,
+      resultCountsPerMediaType,
+      resultCount,
+      fetchState,
+      allMedia,
+
+      fetchSingleMediaType,
+      fetchMediaItem,
+      fetchMedia,
+      clearMedia,
+
+      // Elements exported only for testing, should not be used by components
+      test: {
+        fetchStartMedia,
+        fetchEndMedia,
+        fetchMediaError,
+        setMedia,
+        mediaNotFound,
+        handleMediaError,
+      },
```

Comment:
The fetch state stuff seems like it could be it's own composable btw, in fact, it _is_ through `useFetch` and `useAsyncData` from Nuxt, right?

Reply:
I am hesitant to use `useFetch`  because it's not used in the same way in Nuxt 3.

I like the idea of creating a new `useFetchState` composable. I was thinking about using a status enum (idle/success/error/fetching), and error ref (null/error message), and a computed that returns the `isFetching/fetchingError` object.

Reply:
That sounds good to me! Could that be in a separate PR that blocks this one?

Reply:
Resolved by the creation of `useFetchState` in #1178